apps/web/vibes/soul/primitives/navigation/index.tsx

```diff
@@ -861,3 +883,61 @@ function LocaleForm({
     </DropdownMenu.Root>
   );
 }
+
+function CurrencyForm({
+  action,
+  currencies,
+  activeCurrencyId,
+}: {
+  activeCurrencyId?: string;
+  action: CurrencyAction;
+  currencies: [Currency, ...Currency[]];
+}) {
+  const [lastResult, formAction] = useActionState(action, null);
+  const activeCurrency = currencies.find((currency) => currency.id === activeCurrencyId);
+
+  useEffect(() => {
+    if (lastResult?.error) console.log(lastResult.error);
```

Comment:
We should use alerts here together with `useForm` from conform to avoid reaching into `lastResult` which is meant to be opaque